src/native/libs/System.Native/pal_threading.c

```diff
@@ -121,16 +133,19 @@ void SystemNative_LowLevelMonitor_Destroy(LowLevelMonitor* monitor)
 
     (void)error; // unused in release build
 
+#endif /* !TARGET_WASI */
     free(monitor);
 }
 
 void SystemNative_LowLevelMonitor_Acquire(LowLevelMonitor* monitor)
 {
     assert(monitor != NULL);
 
+#if !defined(TARGET_WASI)
     int error = pthread_mutex_lock(&monitor->Mutex);
     assert(error == 0);
     (void)error; // unused in release build
+#endif /* !TARGET_WASI */
 
     SetIsLocked(monitor, true);
```

Comment:
The ifdefs around SetIsLocked are inconsistent. The call is sometimes outside the ifdef, and sometime inside the ifdef.